jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js

```diff
@@ -21,17 +21,59 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 YUI.add('deployment-bar', function() {
 
   juju.components.DeploymentBar = React.createClass({
+    previousNotifications: [],
 
-  /**
-    Get the label for the deploy button.
+    /**
+      Get the current state of the deployment bar.
 
-    @method _getDeployButtonLabel
-    @param {Boolean} hasCommits Does the env have commits.
-    @returns {String} the label for the deploy button
-  */
-  _getDeployButtonLabel: function(hasCommits) {
-    return hasCommits ? 'Commit changes' : 'Deploy changes';
-  },
+      @method getInitialState
+      @returns {Object} The current state.
+    */
+    getInitialState: function() {
+      return {
+        latestChangeDescription: null
+      };
+    },
+
+    componentDidMount: function() {
+      this._updateLatestChange(this.props.currentChangeSet);
+    },
+
+    componentWillReceiveProps: function(nextProps) {
```

Comment:
Hi @huwshimi and @kadams54, the shallowRenderer utils method actually supports this. I have written the test for you which updates and calls the `componentWillRecieveProps` method. Be sure to read the inline notes which explain how this works. Hope it helps :)

```JavaScript
  fit('updates latest change on new props', function() {
    var currentChangeSet = {one: 1, two: 2};
    var deployButtonAction = sinon.stub();
    var generateChangeDescription = sinon.stub();
    generateChangeDescription.returns('changesList');
    // Note that this is returning the renderer here and not the output.
    var renderer = jsTestUtils.shallowRender(
      <juju.components.DeploymentBar
        currentChangeSet={currentChangeSet}
        deployButtonAction={deployButtonAction}
        generateChangeDescription={generateChangeDescription} />,
      // This will cause the util to return the renderer and not the output
      // which you can then call again with the component markup for it to
      // 'update' the component.
      true);
    var output = renderer.getRenderOutput();
    // generateChangeDescription is only called if it's been updated.
    assert.equal(generateChangeDescription.callCount, 0);
    assert.deepEqual(output,
      <juju.components.Panel instanceName="deployment-bar-panel" visible={true}>
        <juju.components.DeploymentBarNotification change={null} />
        <juju.components.DeploymentBarChangeCount count={2} />
        <juju.components.GenericButton
          action={deployButtonAction}
          type="confirm"
          disabled={false}
          title="Deploy changes" />
      </juju.components.Panel>);
    // If you put a console.log in both the componentDidMount and
    // componentWillReceiveProps methods you'll see that the componentDidMount
    // method is never called (as expected), but the componentWillReceiveProps
    // is called on the second render.
    var currentChangeSet = {one: 1, two: 2, three: 3};
    renderer.render(
      <juju.components.DeploymentBar
        currentChangeSet={currentChangeSet}
        deployButtonAction={deployButtonAction}
        generateChangeDescription={generateChangeDescription} />);

    output = renderer.getRenderOutput();
    assert.deepEqual(output,
      <juju.components.Panel instanceName="deployment-bar-panel" visible={true}>
        // Note that "changesList" is what was returned from the
        // generateChangeDescription stub defined above.
        <juju.components.DeploymentBarNotification change="changesList" />
        // Note that the count has updated.
        <juju.components.DeploymentBarChangeCount count={3} />
        <juju.components.GenericButton
          action={deployButtonAction}
          type="confirm"
          disabled={false}
          title="Deploy changes" />
      </juju.components.Panel>);
  });
```